docs/docs/configuration/overview.md

```diff
@@ -170,6 +170,7 @@ An example [oauth2-proxy.cfg](https://github.com/oauth2-proxy/oauth2-proxy/blob/
 | `--reverse-proxy` | bool | are we running behind a reverse proxy, controls whether headers like X-Real-IP are accepted and allows X-Forwarded-{Proto,Host,Uri} headers to be used on redirect selection | false |
 | `--scope` | string | OAuth scope specification | |
 | `--session-cookie-minimal` | bool | strip OAuth tokens from cookie session stores if they aren't needed (cookie session store only) | false |
+| `--session-expiry-offset` | duration | oauth2-proxy will try to refresh the session this amount of time before it expires | 5m |
```

Comment:
Not sure how I feel about making this on by default, if we had the default as 0, this would mean that this feature is opt in and we would retain existing behaviour. Is it worth having this off by default and adding some details about this option to the docs instead?

Reply:
I think 0 would mean that there would be no offset and the proxy would try the refresh only after the token has expired. But this might be a problem if the token will expire soon and the access token is passed forward the protected service. The token might be already expired when the protected service inspects the token. (Maybe also because the server clocks might be slightly desync.)

I am also not sure why the old behavior should still be supported in the future. This means that the session is still maintained although it is expired. Why would this be intended?

Reply:
> I think 0 would mean that there would be no offset and the proxy would try the refresh only after the token has expired.

I think looking at the code, 0 retains the old behaviour right?

> But this might be a problem if the token will expire soon and the access token is passed forward the protected service.

Agreed, this is a probably for _some_ providers, but not all. This is why I would rather this be an opt in feature rather than opt out. For example, I've never had an issue with this when working with ID Tokens and refresh tokens using Dex and Google, the ID Token and refresh were all set to the same period and the proxy refreshed when it needed to. I think rather than forcing everyone into this, they can opt in if they need it.

> This means that the session is still maintained although it is expired. Why would this be intended?

This is not necessarily true, as soon as we notice the expired session in the current behaviour, we attempt to refresh, it just may not be that the session timing is based on the access token, it may be on the ID Token instead.

I guess this may be desired to make sure sessions are only refreshed right on expiry :thinking: 

I guess another reason not to set a default is that we don't know how long users sessions are going to be. If the default early refresh is 5 minutes and a users session is only 10 minutes, that's a significant portion. If the session is 1 minute (I've seen people doing this in testing), then they would constantly refresh on every request. I think users need to be aware of this option, rather than it being magic that just works (some of the time)

Reply:
I think for now, we should make the default 0, but add a todo to make it non-zero on the next breaking release.

IMO changing the behaviour so that we refresh earlier than we do now could be considered a breaking change (someone suddenly realises their sessions breaking earlier than before perhaps?). 

The other question I have re a default, and this doesn't matter so much if we make it zero for now, Is how to choose a good default, 5m seems sensible for a session that's an hour, but maybe it would be more if the session was more